securesystemslib/signer/_signer.py

```diff
@@ -3,15 +3,49 @@
 import logging
 import os
 from abc import ABCMeta, abstractmethod
-from typing import Any, Callable, Dict, Optional, Type
+from typing import Any, Callable, Dict, Optional, Type, cast
 from urllib import parse
 
 import securesystemslib.keys as sslib_keys
```

Comment:
As a note (unrealted to this PR), I was wondering what was still keeping this import in:
We call `decrypt_key()` for the encrypted private key use case (there's also a matching `securesystemslib.keys.encrypt_key()` call).

I'm sure cryptography has something we could use instead?

Reply:
> I'm sure cryptography has something we could use instead?

Yes, it has, and #604 uses it consistently for all 3 supported key types to import encrypted private key files from a standard PEM format simply by...

https://github.com/secure-systems-lab/securesystemslib/blob/a1e280fbb9751c5bfc286e0e3f7c96df9debf0b8/securesystemslib/signer/_signer.py#L305

#604 does not yet provide API to serialise and dump corresponding private keys to disk but that would be extremely easy to to add, using `CryptoSigner.private_key.private_bytes` as [described](https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#key-serialization).

`securesystemslib` really shouldn't implement any more complex encryption or decryption logic than this.